[DOC] Update Set up test app to use agent flow (#3228)
grafana/tempo: docs/sources/tempo/setup/set-up-test-app.md

You'll need:

* Grafana 10.0.0 or higher
* Microservice deployments require the Tempo querier URL, for example: `http://tempo-cluster-query-frontend.tempo.svc.cluster.local:3100/`
* [OpenTelemetry telemetrygen](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/cmd/telemetrygen) for generating tracing data

Refer to [Deploy Grafana on Kubernetes](/docs/grafana/latest/setup-grafana/installation/kubernetes/#deploy-grafana-on-kubernetes) if you are using Kubernetes.
Otherwise, refer to [Install Grafana](/docs/grafana/latest/installation/) for more information.

## Configure Grafana Agent Flow to remote-write to Tempo

> We'll use a [Grafana Agent Helm chart](https://grafana.com/docs/agent/latest/flow/setup/install/kubernetes/) deployment to send traces to Tempo.

Author: ```suggestion
We'll use a [Grafana Agent Helm chart](/docs/agent/latest/flow/setup/install/kubernetes/) deployment to send traces to Tempo.
```

To do this, you need to create a configuration that can be used by the Agent to receive and export traces in OTLP protobuf format.

> Create a new `values.yaml` file which we'll use as part of the Agent install.

Author: ```suggestion
1. Create a new `values.yaml` file which we'll use as part of the Agent install.
```

1. Create the file by using `touch`.
   ```bash

>    touch values.yaml

Author: I added the "Create the file..." sentence as an interpretation of the actual command. Maybe I got it wrong? 

@hedss is it reasonable to say Create a `values.yaml` file or open an existing one used with Tempo. and then delete touch values.yaml?

> 1. Create the file by using `touch`.
>    ```bash
>    touch values.yaml

Author: ```suggestion
1. Create a values.yaml file or open an existing one used with Tempo
```

>    touch values.yaml

Reviewer: Yep, we should do this.

>    ```

Author: ```suggestion
```

                     // The connection is insecure.
                     insecure = true
                     // Do not verify TLS certificates when connecting.
                     insecure_skip_verify = true
                 }
             }
         }
   ```
   Ensure that you use the specific namespace you've installed Tempo in for the OTLP exporter. In the line:
   ```yaml
   endpoint = "http://tempo-cluster-distributor.tempo.svc.cluster.local:3100"
   ```
   change `tempo` to reference the namespace where Tempo is installed, for example:  `http://tempo-cluster-distributor.my-tempo-namespaces.svc.cluster.local:3100`.

1. Deploy the Agent using Helm:
   ```bash
   helm install -f values.yaml grafana-agent grafana/grafana-agent
   ```
   If you wish to deploy the agent into a specific namespace, make sure to create the namespace first and specify it to Helm by appending `--namespace=<grafana-agent-namespace>` to the end of the command.

## Create a Grafana Tempo data source

To allow Grafana to read traces from Tempo, you must create a Tempo data source.

1. Navigate to **Connections** > **Data Sources**.

1. Click on **Add data source**.

1. Select **Tempo**.

> 1. Set the URL to `http://<TEMPO-DISTRIBUTOR-SERVICE>:<HTTP-LISTEN-PORT>/`, filling in the path to Tempo's query frontend service, and the configured HTTP API prefix. If you have followed the [Deploy Tempo with Helm installation example]({{< relref "../setup/helm-chart.md" >}}), the query frontend service's URL will look something like this: `http://tempo-cluster-query-frontend.<namespace>.svc.cluster.local:3100`

Author: Good catch. We changed from distributor to query while we were testing. Looks like we missed this. I'll fix it. 
```suggestion
1. Set the URL to `http://<TEMPO-QUERY-FRONTEND-SERVICE>:<HTTP-LISTEN-PORT>/`, filling in the path to Tempo's query frontend service, and the configured HTTP API prefix. If you have followed the [Deploy Tempo with Helm installation example]({{< relref "../setup/helm-chart.md" >}}), the query frontend service's URL will look something like this: `http://tempo-cluster-query-frontend.<namespace>.svc.cluster.local:3100`
```

> 1. Set the URL to `http://<TEMPO-DISTRIBUTOR-SERVICE>:<HTTP-LISTEN-PORT>/`, filling in the path to Tempo's query frontend service, and the configured HTTP API prefix. If you have followed the [Deploy Tempo with Helm installation example]({{< relref "../setup/helm-chart.md" >}}), the query frontend service's URL will look something like this: `http://tempo-cluster-query-frontend.<namespace>.svc.cluster.local:3100`

Reviewer: Yeah, this was snow-blindness, we'd changed it in the example but not the main text. :)

You'll need:

* Grafana 10.0.0 or higher
* Microservice deployments require the Tempo querier URL, for example: `http://tempo-cluster-query-frontend.tempo.svc.cluster.local:3100/`
* [OpenTelemetry telemetrygen](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/cmd/telemetrygen) for generating tracing data

Refer to [Deploy Grafana on Kubernetes](/docs/grafana/latest/setup-grafana/installation/kubernetes/#deploy-grafana-on-kubernetes) if you are using Kubernetes.
Otherwise, refer to [Install Grafana](/docs/grafana/latest/installation/) for more information.

## Configure Grafana Agent Flow to remote-write to Tempo

We'll use a [Grafana Agent Helm chart](/docs/agent/latest/flow/setup/install/kubernetes) deployment to send traces to Tempo.

To do this, you need to create a configuration that can be used by the Agent to receive and export traces in OTLP protobuf format.

> 1. Create a new `values.yaml` file which we'll use as part of the Agent install.
> 
> 1. Create a new `values.yaml` file which we'll use as part of the Agent install.

Author: ```suggestion
1. Create a new `values.yaml` file which we'll use as part of the Agent install.

```

Next, you can use [OpenTelemetry `telemetrygen`](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/cmd/telemetrygen) to generate tracing data to test your Tempo installation.

In the following instructions we assume the endpoints for both the Grafana Agent and the Tempo distributor are those described above, for example:
* `grafana-agent.grafana-agent.svc.cluster.local` for Grafana Agent

> * `tempo-cluster-distributor.tempo.svc.cluster.local` for the Tempo distributor

Author: ```suggestion
  * `tempo-cluster-query-frontend.tempo.svc.cluster.local` for the Tempo distributor
```

> * `tempo-cluster-distributor.tempo.svc.cluster.local` for the Tempo distributor

Reviewer: This isn't for querying though, this is for pushing traces directly at Tempo, so we need the distributor. Am I missing something here?

> * `tempo-cluster-distributor.tempo.svc.cluster.local` for the Tempo distributor

Reviewer 2: yup. agree. this is the write path. i got confused on this one.
